AndroidTrainingModule/src/com/cyscorpions/dalejulian/androidtrainingmodule/MainActivity.java

```diff
@@ -1,35 +1,97 @@
 package com.cyscorpions.dalejulian.androidtrainingmodule;
 
-import android.support.v7.app.ActionBarActivity;
+import java.util.Date;
+
+import com.cyscorpions.dalejulian.androidtrainingmodule.DatePickerFragment.ReceiveDateDialogListener;
+
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.app.FragmentManager;
 import android.os.Bundle;
 import android.view.Menu;
 import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+public class MainActivity extends FragmentActivity implements ReceiveDateDialogListener {
+
+	private Date mDate = new Date();
+	private TextView mTextView;
+	private Button mChangeButton;
+	private Button mHelloWorldButton;
+	private Button mShowDatePickerButton;
+	private EditText mEditText;
+
+	private static final String DIALOG_DATE = "date";
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.relative_layout_activity);
+
+		mTextView = (TextView) findViewById(R.id.main_textView);
+		mChangeButton = (Button) findViewById(R.id.main_changeGreet_button);
+		mEditText = (EditText) findViewById(R.id.main_editText);
+		mHelloWorldButton = (Button) findViewById(R.id.main_displayHelloWorld_button);
+		mShowDatePickerButton = (Button) findViewById(R.id.main_showDatePicker_button);
+
+		mChangeButton.setOnClickListener(new View.OnClickListener() {
```

Comment:
As for Android Best Practice, it is recommended to limit the scope of every variable. You may directly use `findViewById(R.id.main_changeGreet_button).setOnClickListener(clickListener)` since they were only used for click listener. Same as for other buttons.